### ISSUE TITLE
SIMD not enabled in Eigen

### ISSUE THREAD

**Issue by forderud** [forderud/QtWasm#68]
Observed output when running the sample project:
![image](https://github.com/forderud/QtWasm/assets/2671400/c68c5543-990c-4274-bff3-be229b79ab9d)

The `Eigen version 3.4.0 (SIMD: None)` console output means that **SIMD is not enabled in Eigen**.

### Relevant resources
* [Using SIMD with WebAssembly](https://emscripten.org/docs/porting/simd.html)
* Eigen [ConfigureVectorization.h](https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Core/util/ConfigureVectorization.h) sources. This file may need to be updated.

## Already tried
I've already tried to add the following to the project CMakeLists.txt. However, it doesn't help.
```
# Enable 128bit SIMD
add_compile_options(-msimd128)
```

## Eigen version installed
```
# apt show libeigen3-dev
Package: libeigen3-dev
Version: 3.4.0-2ubuntu2
Priority: extra
Section: universe/libdevel
Source: eigen3
Origin: Ubuntu
...
```

Eigen 3.4.0 appear to be the latest released version according to https://gitlab.com/libeigen/eigen/-/tags?sort=updated_desc

**Comment by qdel** [comment]
I had a look at it. In fact we have multiple issues here:

- first the simd128 flag enables auto vectorization. So performance gains depends on how the compiler detect how the code can be optimized. The advantage is that we barely have risk here enabling it (regressions / browser support).
- To enable vectorization by eigen, we need at list to pass `-msse2` to the compiler. emscripten default compiler does not enable any explicit vectorization support, (while gcc defaults to sse2 for example)
 - When enabling -msse2 flag, we met first issue: the code of tag 3.4.0 of eigen include wrong file: https://gitlab.com/libeigen/eigen/-/blob/3147391d946bb4b6c68edd901f2add6ac1f31f8c/Eigen/src/Core/util/ConfigureVectorization.h#L345 (this is the version actually released by ubuntu), This is fixed on the 3.4.0 **branch**: https://gitlab.com/libeigen/eigen/-/blob/3.4/Eigen/src/Core/util/ConfigureVectorization.h?ref_type=heads#L345
    - But we will find next issue: https://gitlab.com/libeigen/eigen/-/issues/2742. I will make a follow up on this ticket...
    - Switching to master fix the issue !
 
![image](https://github.com/forderud/QtWasm/assets/1213231/84168ad8-f5d6-4129-86cb-d56722d43b21)

(yes, I went to full range...)

**Comment by forderud** [comment]
Problem fixed by https://github.com/forderud/QtWasm/pull/70
